spring-jdbc/src/test/java/org/springframework/jdbc/core/simple/SimpleJdbcInsertTests.java

```diff
@@ -81,4 +82,36 @@ public void testNoSuchTable() throws Exception {
 		verify(resultSet).close();
 	}
 
+	@Test
+	public void testSimpleJdbcInsert() {
+		SimpleJdbcInsert jdbcInsert = new SimpleJdbcInsert(dataSource).withTableName("T").usingColumns("F", "S");
+		jdbcInsert.compile();
+		String expected = "INSERT INTO T (F, S) VALUES(?, ?)";
+		String actual = jdbcInsert.getInsertString();
+		assertThat(actual).isEqualTo(expected);
+	}
+
+	@Test
+	public void testSimpleJdbcInsertWithEscapingWithSchemaName() throws Exception {
+		SimpleJdbcInsert jdbcInsert = new SimpleJdbcInsert(dataSource).withSchemaName("S").withTableName("T").usingColumns("F", "S").usingEscaping(true);
+
+		given(databaseMetaData.getIdentifierQuoteString()).willReturn("`");
+
+		jdbcInsert.compile();
+		String expected = "INSERT INTO `S.T` (`F`, `S`) VALUES(?, ?)";
```

Comment:
```suggestion
		String expected = "INSERT INTO `S`.`T` (`F`, `S`) VALUES(?, ?)";
```

Just a side note: the schema and table names have to be quoted independently.

I've fixed this in my local branch and added integration tests with H2 to verify it.